src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -633,11 +626,32 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
     }
 
 
+
+  let sourceFileToProjectOptions =
+    aval {
+      let! options = loadedProjectOptions
+      return
+        options
+        |> List.collect(fun proj -> proj.SourceFiles |> Array.map(fun source -> Utils.normalizePath source, proj) |> Array.toList)
+        |> List.groupByFst
+
+    }
+    |> AMap.ofAVal
   let fantomasLogger = LogProvider.getLoggerByName "Fantomas"
   let fantomasService: FantomasService = new LSPFantomasService() :> FantomasService
 
-  let openFilesTokens = cmap<string<LocalPath>, cval<CancellationTokenSource>> ()
+  let openFilesTokens =
+    ConcurrentDictionary<string<LocalPath>, CancellationTokenSource>()
```

Comment:
Changed to be outside the Adaptive world as it's better to reset this information without notifying dependents of changes to recompute.